capsules/src/net/ipv6/ipv6_send.rs

```diff
@@ -0,0 +1,158 @@
+use core::cell::Cell;
+use ieee802154::device::{MacDevice, TxClient};
+use kernel::ReturnCode;
+use kernel::common::take_cell::TakeCell;
+use net::ieee802154::MacAddress;
+use net::ipv6::ip_utils::IPAddr;
+use net::ipv6::ipv6::{IP6Header, IP6Packet, TransportHeader};
+use net::sixlowpan::sixlowpan_state::TxState;
+
+// TODO: These should *not* be constants, and should be set at some other
+// point during the initialization of the IP stack
+const SRC_MAC_ADDR: MacAddress = MacAddress::Short(0xf00f);
+const DST_MAC_ADDR: MacAddress = MacAddress::Short(0xf00e);
+
+pub trait IP6Client {
+    fn send_done(&self, result: ReturnCode);
+}
+
+pub trait IP6Sender<'a> {
```

Comment:
Part of the reason for that is that the current design has the IP6Sender implementation owning the `IP6Packet` structure/buffer. So we no longer pass in an entire packet to the IP6Sender, but just headers plus a payload, which the IP6Sender layer converts to an `IP6Packet`. I think we should change the design to have a `send` method that takes in an arbitrary `IP6Header`, but the current design doesn't support sending a full `IP6Packet`.

Part of the reason for this ambiguity is that we weren't sure of buffer ownership, and so we just decided to have the `IP6Sender` layer own the `IP6Packet` buffer. This will probably change as we get a better idea of the upper layers.